src/popup/utils/utils.ts

```diff
@@ -401,12 +400,18 @@ export function amountRounded(rawAmount: number | BigNumberPublic): string {
   return amount.toFixed((amount < 0.01) ? 9 : 2);
 }
 
-export function getTxType(tx: ITx): TxType {
-  return (
-    TX_TYPE_MDW[tx.type]
-    || (tx.tag && SCHEMA.OBJECT_ID_TX_TYPE[tx.tag])
-    || (Object.values(SCHEMA.TX_TYPE).includes(tx.type) && tx.type)
-  );
+export function getTxType(tx: ITx): Tag | null {
+  const typeKey = tx.type as keyof typeof Tag;
```

Comment:
Is it possible to `tx.type` to have a proper type from the beginning?
Also this `keyof typeof Tag` type is used more than once. Maybe it would be good to create separate type from this?